### ISSUE TITLE
twisted.internet.kqreactor should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7823]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7823|
|Type|enhancement|
|Created|2015-03-18 03:19:42Z|
|Branch|https://github.com/twisted/twisted/tree/kqueue-python3-7823|

<details><summary>Searchable metadata</summary>

```
trac-id__7823 7823
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_kqueue_python3_7823 branches/kqueue-python3-7823
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1426648782573609 1426648782573609
changetime__1426863379134911 1426863379134911
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [44108]) Branching to kqueue-python3-7823.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Done the porting work, added to dist3, tested under KQueue on Py3.3 locally (OS X 10.10) and got passes.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Current changes look good, but do we have tests for kqreactor ? I don't see them in dist3.py so I don't know how these changes are validated by the py3.3 builder?

What tests can run on py3 to see the code from trunk fail and then seeing them pass on this branch?

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

They're not validated by the 3.3 builder -- because we don't have a BSD Py3.3 builder yet.

You can test this by adding the following to run_python3_tests before it runs unittest on a BSD (FreeBSD or OSX):

```
from twisted.internet import kqreactor
kqreactor.install()
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Ugly business :(

then it is good to merge.


Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44167]) Merge kqueue-python3-7823: twisted.internet.kqreactor should be ported to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7823](https://github.com/twisted/twisted/issues/7823)
